File: 724-find-pivot-index/find-pivot-index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        pre=0
        for i in nums:
            pre+=i
        now=0
        for j in range(len(nums)):
            if now==pre-now-nums[j]:
                return j
            now+=nums[j]
        return -1

File: 724-find-pivot-index/find-pivot-index.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        int total=0;
        for (auto i : nums){
            total+=i;
        }
        int pre=0;
        int n=nums.size();
        for (int i = 0; i<n; i++){
            int now=total-pre-nums[i];
            if (now==pre){
                return i;
            }
            pre=pre+nums[i];
        }
        return -1;
    }
};